ballerina/tests/user_client.bal

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerinax/mysql;
+import ballerina/sql;
+
+client class UserClient {
+
+    private final string entityName = "User";
+    private final sql:ParameterizedQuery tableName = `Users`;
+    private final map<FieldMetadata> fieldMetadata = {
+        id: {columnName: "id", 'type: int},
+        name: {columnName: "name", 'type: string},
+        "profile.id": {'type: int, relation: {entityName: "profile", refTable: "Profiles", refField: "id"}},
+        "profile.name": {'type: string, relation: {entityName: "profile", refTable: "Profiles", refField: "name"}}
+    };
+    private string[] keyFields = ["id"];
+    private final map<JoinMetadata> joinMetadata = {
+        profile: {refTable: "Profiles", refFields: ["userId"], joinColumns: ["id"]}
+    };
+
+    private SQLClient persistClient;
+
+    public function init() returns error? {
+        mysql:Client dbClient = check new (host = HOST, user = USER, password = PASSWORD, database = DATABASE, port = PORT);
+        self.persistClient = check new (self.entityName, self.tableName, self.fieldMetadata, self.keyFields, dbClient, self.joinMetadata);
+    }
+
+    remote function create(User value) returns User|error {
+        sql:ExecutionResult result = check self.persistClient.runInsertQuery(value);
+        
+        int key;
+        if result.lastInsertId is () {
+            key = value.id;
+        } else {
+            key = <int>result.lastInsertId;
+        }
+
+        User user = {
+            id: key,
+            name: value.name
+        };
+
+        if value.profile is Profile {
+            ProfileClient profileClient = check new ProfileClient();
+            Profile profile = <Profile>value.profile.clone();
+            profile.user = user;
+            boolean exists = check profileClient->exists(<Profile>value.profile);
+            if !exists {
+                user.profile = check profileClient->create(profile);
+            } else {
+                check profileClient->update({"user.id": user.id}, {id: profile.id});
+            }
```

Comment:
Shall we support creating/updating/deleting associated records only via owner? Seems like the other way is a bit complex. even hibernate supports only via the owner. WDYT?

Reply:
Fixed in [8a80d26](https://github.com/ballerina-platform/module-ballerina-persist/pull/11/commits/8a80d263be98cd46f659f4c7ee8429cce0848163)

Reply:
Fixed in [1131dcb](https://github.com/ballerina-platform/module-ballerina-persist/pull/11/commits/1131dcba3866772892a69fef1b41de95bd433776)